mm/kfence/core.c

```diff
@@ -308,7 +314,7 @@ void *kfence_guarded_alloc(struct kmem_cache *cache, size_t override_size,
 		kfence_metadata[index].alloc_stack =
 			save_stack(gfp & ~__GFP_RECLAIM);
 		kfence_metadata[index].cache = cache;
-		kfence_metadata[index].size = right ? -size : size;
+		WRITE_ONCE(kfence_metadata[index].size, right ? -size : size);
```

Comment:
Just FYI, we can now run KCSAN on KFENCE. ;-)

This is more an AI for me, when I get back to reworking the test into more of a torture test.

Reply:
ack